File: ldk/node/src/hostClients/uiClient.ts
import BaseClient, { GRPCClientConstructor } from './baseClient';
import { UIClient as UIGRPCClient } from '../grpc/ui_grpc_pb';
import { UIService } from './uiService';
import { StreamListener, StoppableStream } from './stoppableStream';
import Messages from '../grpc/ui_pb';
import { TransformingStream } from './transformingStream';

export class UIClient extends BaseClient<UIGRPCClient> implements UIService {
  protected generateClient(): GRPCClientConstructor<UIGRPCClient> {
    return UIGRPCClient;
  }

  streamSearchbar(listener: StreamListener<string>): StoppableStream<string> {
    return new TransformingStream<Messages.SearchbarStreamResponse, string>(
      this.client.searchbarStream(
        new Messages.SearchbarStreamRequest().setSession(
          this.createSessionMessage(),
        ),
      ),
      (message) => {
        return message.getText();
      },
      listener,
    );
  }

  streamGlobalSearch(
    listener: StreamListener<string>,
  ): StoppableStream<string> {
    return new TransformingStream<Messages.GlobalSearchStreamResponse, string>(
      this.client.globalSearchStream(
        new Messages.GlobalSearchStreamRequest().setSession(
          this.createSessionMessage(),
        ),
      ),
      (message) => {
        return message.getText();
      },
      listener,
    );
  }
}

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/UI/UIClient.cs
using System.Threading;
using Grpc.Core;
using OliveHelpsLDK.Logging;
using Proto;

namespace OliveHelpsLDK.UI
{
    internal class UIClient : BaseClient<Proto.UI.UIClient>, IUIService
    {
        internal UIClient(CallInvoker channelBase, Session session, ILogger logger) : base(
            new Proto.UI.UIClient(channelBase), session, logger, "ui")
        {
        }

        internal UIClient(Proto.UI.UIClient client, Session session, ILogger logger) : base(
            client, session, logger, "ui")
        {
        }

        public IStreamingCall<string> StreamGlobalSearch(CancellationToken cancellationToken = default)
        {
            var request = new GlobalSearchStreamRequest
            {
                Session = CreateSession()
            };
            var call = Client.GlobalSearchStream(request, CreateOptions(cancellationToken));
            var loggedParser = LoggedParser<GlobalSearchStreamResponse, string>(response => response.Text);
            return new StreamingCall<GlobalSearchStreamResponse, string>(call, loggedParser);
        }

        public IStreamingCall<string> StreamSearchbar(CancellationToken cancellationToken = default)
        {
            var request = new SearchbarStreamRequest
            {
                Session = CreateSession()
            };
            var call = Client.SearchbarStream(request, CreateOptions(cancellationToken));
            var loggedParser = LoggedParser<SearchbarStreamResponse, string>(response => response.Text);
            return new StreamingCall<SearchbarStreamResponse, string>(call, loggedParser);
        }
    }
}